test/test_helper.rb

```diff
@@ -39,19 +39,24 @@ def setup
   end
 
   def launch(protocol: 'https', plugins: [], settings: {})
-    port = 1024 + rand(63_000)
+    port = 0
     @settings = Proxy::Settings::Global.new(settings.merge("#{protocol}_port" => port))
     @t = Thread.new do
       launcher = Proxy::Launcher.new(@settings)
       app = launcher.public_send("#{protocol}_app", port, plugins)
-      launcher.webrick_server(app.merge(AccessLog: [Logger.new('/dev/null')]), ['localhost'], port).start
+      @server = server = launcher.webrick_server(app.merge(AccessLog: [Logger.new('/dev/null')]), ['localhost'], port)
```

Comment:
Oops, debug left over. Will update:
```suggestion
      server = launcher.webrick_server(app.merge(AccessLog: [Logger.new('/dev/null')]), ['localhost'], port)
```